PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/ruleConfigs.json

```diff
@@ -0,0 +1,7 @@
+{
+  "ConnectionStringCheck": {
+    "method_to_check": "connectionString",
+    "antipattern_message" : "Connection String detected. Use DefaultAzureCredential for azure service client authentication instead.",
+    "solution": "Do not construct a service client with a connection string unless such connection string is available within the Azure portal or Azure CLI. Use DefaultAzureCredential or other secure methods for client authentication instead."
```

Comment:
@g2vinay is this the correct recommendattion?
Or do we want to ask users to use DefaultAzureCredential even if connection strings authentication is still available?

Reply:
Yeah, we should rephrase to say, DefaultAzureCredential is recommended if the service client supports Token Credential (Entra ID Authentication). if not, then use Azure Key Credential / Connection Strings based authentication.